src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -672,8 +680,12 @@ private bool AnyProjectHasVulnerablePackageWarning(IReadOnlyList<RestoreSummary>
                 }
 
                 _packageCount += packages.Count;
-                var missingPackagesList = packages.Where(p => p.IsMissing).ToList();
-                _missingPackagesCount = missingPackagesList.Count;
+                _missingPackagesCount = 0;
```

Comment:
This is a non-allocation way of calculating the missing packages count.
Previously we were allocating a list we never used.